openstates/oh/bills.py

```diff
@@ -13,43 +13,333 @@
+                                                    action_desc,
+                                                    date,
+                                                    type = action_type,
+                                                    committees = committees)
+
+                            self.add_document(all_amendments, bill_id,"amendment",bill,first_page)
+                            self.add_document(all_fiscals,bill_id,"fiscal",bill,first_page)
+                            self.add_document(all_synopsis,bill_id,"synopsis",bill,first_page)
+                            self.add_document(all_analysis,bill_id,"analysis",bill,first_page)
+
+                            #todo: votes, committee votes
+                            vote_url = base_url+bill_version["votes"][0]["link"]
+                            vote_doc = self.get(vote_url)
+                            votes = vote_doc.json()
+                            self.process_vote(votes,vote_url,bill,legislators,chamber_dict,vote_results)
+
+                            vote_url = base_url+bill_version["cmtevotes"][0]["link"]
+                            vote_doc = self.get(vote_url)
+                            votes = vote_doc.json()
+                            self.process_vote(votes,vote_url,bill,legislators,chamber_dict,vote_results)
+
+
+                            #we have never seen a veto or a disapprove, but they seem important.
+                            #so we'll check and throw an error if we find one
+                            #life is fragile. so are our scrapers.
+                            if "veto" in bill_version:
+                                veto_url = base_url+bill_version["veto"][0]["link"]
+                                veto_json = self.get(veto_url).json()
+                                if len(veto_json["items"]) > 0:
+                                    raise AssertionError("Whoa, a veto! We've never gotten one before. Go write some code to deal with it: {}".format(veto_url))
+
+                            if "disapprove" in bill_version:
+                                disapprove_url = base_url+bill_version["disapprove"][0]["link"]
+                                disapprove_json = self.get(disapprove_url).json()
+                                if len(disapprove_json["items"]) > 0:
+                                    raise AssertionError("Whoa, a disapprove! We've never gotten one before. Go write some code to deal with it: {}".format(disapprove_url))
+
+
+
+
+                        #this stuff is version-specific
+                        version_name = bill_version["version"]
+                        version_link = base_url+bill_version["pdfDownloadLink"]
+                        mimetype = "application/pdf" if version_link.endswith("pdf") else None
+                        bill.add_version(version_name,version_link,mimetype=mimetype)
+
+                    self.save_bill(bill)
+
+    def pages(self,base_url, first_page):
```

Comment:
Well we could change per page and also use that function -- to reduce load.

Nice idea!
On Mar 5, 2015 2:35 PM, "Miles Watkins" <notifications@github.com> wrote:

> In openstates/oh/bills.py
> <https://github.com/sunlightlabs/openstates/pull/602#discussion_r25894061>
> :
>
> > +                                disapprove_json = self.get(disapprove_url).json()
> > +                                if len(disapprove_json["items"]) > 0:
> > +                                    raise AssertionError("Whoa, a disapprove! We've never gotten one before. Go write some code to deal with it: {}".format(disapprove_url))
> > +
> > +
> > +
> > +
> > +                        #this stuff is version-specific
> > +                        version_name = bill_version["version"]
> > +                        version_link = base_url+bill_version["pdfDownloadLink"]
> > +                        mimetype = "application/pdf" if version_link.endswith("pdf") else None
> > +                        bill.add_version(version_name,version_link,mimetype=mimetype)
> > +
> > +                    self.save_bill(bill)
> > +
> > +    def pages(self,base_url, first_page):
>
> What if we just made our API requests with a ?per_page=1000 (default is
> 50) instead of using a pages function? It doesn't seem to be disallowed,
> and it would result in fewer calls to their API and allow the removal of a
> whole method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sunlightlabs/openstates/pull/602/files#r25894061>.
>